aten/src/ATen/native/cpu/UpSampleKernel.cpp

```diff
@@ -758,6 +758,12 @@ struct HelperInterpBase {
     xsize = std::min(
         static_cast<int64_t>(center + support + 0.5 + align_corners_delta), input_size) - xmin;
 
+    if (!antialias && C10_UNLIKELY(xsize > max_interp_size)) {
```

Comment:
Should we clamp regardless of antialias? Even if it never happens when `antialias=True`, it would slightly simplify the logic as we could just do `xsize = std::min(xsize, max_interp_size)` or `xsize = std::clamp(xsize, 0, max_interp_size)`